### ISSUE TITLE
Unsolved type variable when overloading a function with ellipsis as a default

### ISSUE THREAD

**Issue by jack1142** [microsoft/pyright#3958]
**Describe the bug**
When I overload a function taking a TypeVar argument with that argument's default value being an Ellipsis (`...`), I get this warning:
```
Type variable "T" may go unsolved if caller supplies no argument for parameter "default"
  Provide an overload that specifies the return type when the argument is not supplied
```

**To Reproduce**
1. Put code from "Screenshots or Code" into a file.
2. Run pyright on it.
3. See results:
```
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.271
/tmp/example.py
  /tmp/example.py:25:27 - warning: Type variable "T" may go unsolved if caller supplies no argument for parameter "default"
    Provide an overload that specifies the return type when the argument is not supplied (reportInvalidTypeVarUse)
0 errors, 1 warning, 0 informations 
Completed in 0.648sec
```

**Expected behavior**
I expected there to be no warning in at least one of the variants I wrote.

**Screenshots or Code**
I tried two variants.
- Variant number 1:
```python
from types import EllipsisType
from typing import Generic, TypeVar, Union, overload


T = TypeVar("T")


class JsonValue(Generic[T]):
    def __init__(self, value: T) -> None:
        self.value = value


@overload
def get(self) -> JsonValue[int]:
    ...

@overload
def get(self, *, default: EllipsisType) -> JsonValue[int]:
    ...

@overload
def get(self, *, default: T) -> JsonValue[T]:
    ...

def get(self, *, default: T = ...) -> Union[JsonValue[int], JsonValue[T]]:
    if default is ...:
        return JsonValue(123)
    return JsonValue(default)
```

- Variant number 2:
```python
from types import EllipsisType
from typing import Generic, TypeVar, Union, overload


T = TypeVar("T")


class JsonValue(Generic[T]):
    def __init__(self, value: T) -> None:
        self.value = value


@overload
def get(self, *, default: EllipsisType = ...) -> JsonValue[int]:
    ...

@overload
def get(self, *, default: T) -> JsonValue[T]:
    ...

def get(self, *, default: T = ...) -> Union[JsonValue[int], JsonValue[T]]:
    if default is ...:
        return JsonValue(123)
    return JsonValue(default)
```

**VS Code extension or command-line**
I'm running pyright 1.1.271 as a command-line tool.

**Additional context**
N/A

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.272. It will also be included in a future release of pylance.
